File: 1/c/multiples.c
#include <stdio.h>

int main()
{
	int sum = 0;

	int i;

	for (i = 1; i < 1000; i++){
		if (i % 3 == 0)
			sum += i;
		else if (i % 5 == 0)
			sum += i;
	}
	
	printf("%d\n", sum);
}

File: perl/1/multiples.pl
#!/usr/bin/env perl

sub usage {
	die "Usage: multiples.pl [limit]";
}

sub multiple_of_3_or_5 {
	my $num = shift;
	return $num % 3 == 0 || $num % 5 == 0;
}

sub main {
	my $limit = shift;
	my $sum = 0;
	for $i ( 1 .. $limit-1 ) {
		if (multiple_of_3_or_5($i)) {
			$sum += $i;
		}
	}

	print "$sum\n";
}

usage() unless scalar @ARGV;

main(@ARGV);

